flink-1.17-connector-bigquery/flink-connector-bigquery/src/main/java/com/google/cloud/flink/bigquery/sink/serializer/AvroSchemaConvertor.java

```diff
@@ -62,18 +83,23 @@ public static DataType convertToDataType(String avroSchemaString) {
     }
 
     private static DataType convertToDataType(Schema schema) {
+        if (avroToDataTypeConversionMemorizationMap.containsKey(schema)) {
```

Comment:
Instead of doing a double look up; first check if the map contains the key and then do a get, you can simplify the code for just calling get and checking the returned value. If the key does not exist then the return would be a null and if the value is a null then consider that the schema is not cached. 

Also, consider moving the the map put/get behind add/get methods so that later if you want to instrument the cache access to test for efficiency you can easily add it.

Reply:
ACK, Fixed.